### ISSUE TITLE
Make porter agent job retries configurable

### ISSUE THREAD

**Issue by carolynvs** [getporter/operator#149]
Right now we have job retries disabled while developing but it's a core part of k8s and it should be turned on by default. Let's pick a reasonable default to add to the AgentConfig and then allow people to customize how porter retries the agent action jobs.

**Comment by VinozzZ** [comment]
I tried to enable job retries through using the default `BackoffLimit` on the Job spec as well as setting the `RestartPolicy` to `OnFailure` on the PodSpec created by AgentAction job.

A failed AgentAction job was able to retry by restarting the failed pod up to the back off limit before the AgentAction job is considered as Failed. However, the issue I ran into is that all failed pods are deleted after the backoff limit is reached.

I found a good explanation on this here: https://github.com/kubernetes/kubernetes/issues/74848#issuecomment-971487582
Basically if a Job is marked as failure, the pod has to be deleted when `RestartPolicy` is set to `OnFailure` to prevent the pod keeps restarting. Is this an acceptable default behavior that we want?

**Comment by carolynvs** [comment]
> We could keep the RestartPolicy on the pod as Never and have the AgentAction job doing retry by creating new pod and leave the failed ones alone.

I've had good success with that in other projects I have worked on. 👍

> The default retry limit is 6 on a job, maybe we can set it to 3 to reduce the amount of resource consumption?

Let's keep the default but allow people to change the retry limit on the AgentConfig.
